source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1632 @@
+                    const double indx = 1+use_ghost_nodes+(vertex(0) - grid_extent[0].first)/dx;
+
+                    // If our x or y index isn't close to a whole number, then it's likely an artifact
+                    // from using an over-resolved quadrature rule, in that case ignore it.
+                    if (abs(indx - round(indx)) >= precision)
+                      continue;
+
+
+                    // If we're in 2D, we want to take the values and apply them to every row of X points.
+                    if (dim == 2)
+                      {
+                        for (int ys=0; ys<ny; ys++)
+                          {
+                            // FastScape indexes from 1 to n, starting at X and Y = 0, and increases
+                            // across the X row. At the end of the row, it jumps back to X = 0
+                            // and up to the next X row in increasing Y direction. We track
+                            // this to correctly place the variables later on.
+                            // Nx*ys effectively tells us what row we are in
+                            // and then indx tells us what position in that row.
+                            const double index = round(indx)+nx*ys;
+
+                            temporary_variables[0].push_back(vertex(dim-1) - grid_extent[dim-1].second);
+                            temporary_variables[1].push_back(index-1);
+
+                            for (unsigned int i=0; i<dim; ++i)
+                              {
+                                // Always convert to m/yr for FastScape
+                                temporary_variables[i+2].push_back(vel[corner][i]*year_in_seconds);
+                              }
+                          }
+                      }
+                    // 3D case
+                    else
+                      {
+                        // Because indy only gives us the row we're in, we don't need to add 2 for the ghost node.
+                        const double indy = 1+use_ghost_nodes+(vertex(1) - grid_extent[1].first)/dy;
+
+                        if (abs(indy - round(indy)) >= precision)
+                          continue;
+
+                        const double index = round((indy-1))*nx+round(indx);
+
+                        temporary_variables[0].push_back(vertex(dim-1) - grid_extent[dim-1].second);   //z component
+                        temporary_variables[1].push_back(index-1);
+
+                        for (unsigned int i=0; i<dim; ++i)
+                          {
+                            temporary_variables[i+2].push_back(vel[corner][i]*year_in_seconds);
+                          }
+                      }
+                  }
+              }
+
+      return temporary_variables;
+    }
+
+
+    template <int dim>
+    void FastScape<dim>::fill_fastscape_arrays(double *h, double *kd, double *kf, double *vx, double *vy, double *vz, std::vector<std::vector<double>> temporary_variables) const
```

Comment:
Please put each parameter on its own line (same for all other functions in the .cc and .h file). Makes it easier to read.